python/README.md

```diff
@@ -1,41 +1,256 @@
+user, set `PORT4ME_EXCLUDE_SITE`. For example, the systems
+administrator, can choose to exclude an additional set of ports by
+adding the following to file `/etc/profile.d/port4me.sh`:
+
+    ## port4me: always exclude commonly used ports
+    ## https://github.com/HenrikBengtsson/port4me
+
+    PORT4ME_EXCLUDE_SITE=
+
+    ## MySQL
+    PORT4ME_EXCLUDE_SITE=$PORT4ME_EXCLUDE_SITE,3306
+
+    ## ZeroMQ
+    PORT4ME_EXCLUDE_SITE=$PORT4ME_EXCLUDE_SITE,5670
+
+    ## Redis
+    PORT4ME_EXCLUDE_SITE=$PORT4ME_EXCLUDE_SITE,6379
+
+    ## Jupyter
+    PORT4ME_EXCLUDE_SITE=$PORT4ME_EXCLUDE_SITE,8888
+
+    export PORT4ME_EXCLUDE_SITE
+
+In addition to ports excluded via above mechanisms, **port4me** excludes
+ports that are considered unsafe by the Chrome and Firefox web browsers.
+This behavior can be controlled by environment variable
+`PORT4ME_EXCLUDE_UNSAFE`, which defaults to `{chrome},{firefox}`. Token
+`{chrome}` expands to the value of `PORT4ME_EXCLUDE_UNSAFE_CHROME`,
+which defaults to [the set of ports that Chrome
+blocks](https://chromium.googlesource.com/chromium/src.git/+/refs/heads/master/net/base/port_util.cc)
+and `{firefox}` expands to to the value of
+`PORT4ME_EXCLUDE_UNSAFE_FIREFOX`, which defaults to [the set of ports
+that Firefox
+blocks](https://www-archive.mozilla.org/projects/netlib/portbanning#portlist).
+
+Analogously to excluding a set of ports, one can limit the range of
+ports to be scanned by specifying a list or str to the argument `include`, e.g.
+
+    >>> port4me(include=[*range(2000, 2123+1), 4321, *range(10000, 10999+1)])
+    10451
+
+    >>> port4me(include="2000-2123,4321,10000-10999")
+    10451
+
+where the default corresponds to `include = 1024:65535`. Analogously to
+`exclude`, `include` can be specified via environment variables
+`PORT4ME_INCLUDE` and `PORT4ME_INCLUDE_SITE`.
+
+## Scan a predefined set of ports before pseudo-random ones
+
+In addition to scanning the user-specific, pseudo-random port sequence
+for a free port, it is possible to also consider a predefined set of
+ports prior to the random ones by specifying the argument
+`prepend`, e.g.
+
+    >>> port4me(prepend=[4321, 11001], list=5)
+    [4321, 11001, 30845, 19654, 32310]
+
+An alternative to specify them via a command-line option, is to specify
+them via environment variable `PORT4ME_PREPEND`, e.g.
+
+    {alice}$ PORT4ME_PREPEND=4321,11001
+    ...
+    >>> port4me(prepend=[4321, 11001], list=5)
+    [4321, 11001, 30845, 19654, 32310]
+
+The environment variable `PORT4ME_PREPEND` is intended to be used by the
+individual user. To specify a set of ports to be prepended regardless of
+user, set `PORT4ME_PREPEND_SITE`.
+
+## Installation
+
+To install the Python **port4me** package from [PyPI](https://pypi.org/project/port4me/), do:
+
+    {alice}$ pip install port4me
+
+To try it out, call:
+
+    >>> from port4me import port4me
+    >>> port4me("jupyter-notebook")
+    29525
+
+or
+
+    python -m port4me --tool jupyter-notebook
```

Comment:
Does it support the form `<key>=<value>` too, i.e. `python -m port4me --tool=jupyter-notebook`? That's what we use for Bash and eventually also R (I just realized I didn't implement this CLI interface for R).

Reply:
Yeah, it supports both `--tool=xyz` and `--tool xyz`

Reply:
Great, then I'd like to suggest we use `--tool=xyz` in example for consistency.

Reply:
Okay, fixed